third_party/bezierManipulation/src/utils/export.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef SRC_UTILS_EXPORT_HPP
+#define SRC_UTILS_EXPORT_HPP
+
+#include <fstream>
+#include <stdexcept>
+#include <string>
+
+#include "bezierManipulation/src/bezier_spline.hpp"
+#include "bezierManipulation/src/bezier_spline_group.hpp"
+
+namespace beziermanipulation::utils {
+
+/*
+ * Static Class for Export Routines
+ *
+ * Export Routines are extended by whatever format of splines is required.
+ *
+ * Currently supported:
+ *  *.xml -> standard CATS export file (FEAFA, XNS)
+ *  *.itd -> IRIT export file extension
+ */
+class Export {
+ private:
+  /*
+   * Returns a string with a specific file extension
+   *
+   * test.txt and test both return test.txt
+   */
+  static std::string ensureFileExtension(const std::string &filename,
+                                         const std::string &obj_extension);
+
+  /*
+   * Formats spline to IRIT format
+   *
+   * Actual implementation of the spline Export to be called for every spline in
+   * a group Pipes everything directly into a file
+   */
+  template <std::size_t parametric_dimension, typename PhysicalPointType,
+            typename ScalarType>
+  static void format2IRITfile(
+      const BezierSpline<parametric_dimension, PhysicalPointType, ScalarType>
+          &spline,
+      std::ofstream &export_file);
+
+  /*
+   * Formats spline to XML format
+   *
+   * Actual implementation of the spline Export to be called for every spline in
+   * a group Pipes everything directly into a file
+   */
+  template <std::size_t parametric_dimension, typename PhysicalPointType,
+            typename ScalarType>
+  static void format2XMLfile(
+      const BezierSpline<parametric_dimension, PhysicalPointType, ScalarType>
+          &spline,
+      std::ofstream &export_file);
+
+  /*
+   * Formats spline in custom json format
+   *
+   * Actual implementation of the spline Export to be called for every spline in
+   * a group Pipes everything directly into a file
+   */
+  template <std::size_t parametric_dimension, typename PhysicalPointType,
+            typename ScalarType>
+  static void format2JSONfile(
+      const BezierSpline<parametric_dimension, PhysicalPointType, ScalarType>
+          &spline,
+      std::ofstream &export_file);
+
+ public:
+  /// Permit creation of class instance
+  Export() = delete;
+
```

Comment:
do you wanna expose some export functions to python?

Reply:
I was wondering if this is really necessary, because export and import are not really expensive (compared to spline operations). In the newest version of the ugly-named `bezierManipulation`-suite, there is also support for B64 export, if you want -- I let it up to you to decide. All Export Functions are however limited to groups of splines and need to be in `SplineGroup` data structures or single splines.